pkg/node/types/node_test.go

```diff
@@ -273,3 +275,195 @@ func TestClusterServiceValidate(t *testing.T) {
 		})
 	}
 }
+
+func TestGetIPv4AllocCIDRs(t *testing.T) {
+	var (
+		cidr1 = "1.0.0.0/24"
+		cidr2 = "2.0.0.0/24"
+		cidr3 = "3.0.0.0/24"
+	)
+
+	var tests = []struct {
+		// name of test
+		name string
+		// primary ipv4 allocation cidr
+		allocCIDR *string
+		// secondary ipv4 allocation cidrs
+		secAllocCIDRs []*string
+		// expected ipv4 cidrs
+		expect []*string
+	}{
+		{
+			name:          "nil cidrs",
+			allocCIDR:     nil,
+			secAllocCIDRs: nil,
+			expect:        nil,
+		},
+		{
+			name:          "one primary and no secondary cidrs",
+			allocCIDR:     pointer.String(cidr1),
+			secAllocCIDRs: nil,
+			expect: []*string{
+				pointer.String(cidr1),
+			},
+		},
+		{
+			name:      "one primary and one secondary cidr",
+			allocCIDR: pointer.String(cidr1),
+			secAllocCIDRs: []*string{
+				pointer.String(cidr2),
+			},
+			expect: []*string{
+				pointer.String(cidr1),
+				pointer.String(cidr2),
+			},
+		},
+		{
+			name:      "one primary and multiple secondary cidrs",
+			allocCIDR: pointer.String(cidr1),
+			secAllocCIDRs: []*string{
+				pointer.String(cidr2),
+				pointer.String(cidr3),
+			},
+			expect: []*string{
+				pointer.String(cidr1),
+				pointer.String(cidr2),
+				pointer.String(cidr3),
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			var err error
+			var allocCIDR *cidr.CIDR
+
+			if tt.allocCIDR != nil {
+				allocCIDR, err = cidr.ParseCIDR(*tt.allocCIDR)
+				assert.Nil(t, err)
+			}
+
+			var secCIDRs []*cidr.CIDR
+			for _, c := range tt.secAllocCIDRs {
+				if c != nil {
+					secCIDR, err := cidr.ParseCIDR(*c)
+					assert.Nil(t, err)
+					secCIDRs = append(secCIDRs, secCIDR)
+				}
+			}
+
+			expectCIDRs := make([]*cidr.CIDR, 0, len(tt.secAllocCIDRs)+1)
+			for _, c := range tt.expect {
+				if c != nil {
+					expectCIDR, err := cidr.ParseCIDR(*c)
+					assert.Nil(t, err)
+					expectCIDRs = append(expectCIDRs, expectCIDR)
+				}
+			}
```

Comment:
Instead of converting the CIDRs from string here as part of the test, how about using `cidr.MustParseCIDR` in the test case construction above and make the `allocCIDR` and `secAllocCIDRs` members of type `*cidr.CIDR` and the `expect` member `[]*cidr.CIDR` directly?

Reply:
@tklauser thanks for the review. Commit 8c95ae6 implements ^ suggestion.